src/cosmicqc/detection.py

```diff
@@ -0,0 +1,619 @@
+            or bowley_skewness > self.upper_skew_threshold
+        )
+
+    def variability_test_formfactor(self) -> bool:
+        """
+        Calculate Interquartile Range (IQR) score for the FormFactor feature
+        to determine if there is abnormal variability in the shape of the nuclei.
+        More variability in this feature indicates a higher proportion
+        of non-circular nuclei.
+        This can be an indicator of poor segmentations due to non-optimal
+        segmentation parameters or contamination.
+        Threshold is set based on findings from multiple experiments
+        (NF1 & CFReT, see citation file).
+
+        Returns:
+            boolean: True if the distribution is variable, False otherwise.
+        """
+        # use IQR to measure variability of the feature
+        vals = self.dataframe[self.formfactor_feature].dropna()
+        iqr = np.percentile(vals, 75) - np.percentile(vals, 25)
+
+        # determine if distribution is highly variable
+        return iqr > self.variability_threshold
+
+    def check_skew_and_variable(self) -> None:
+        """
+        Step 1: Check for skewness in the cytoplasm texture around the nucleus and
+        variability in the FormFactor of the nucleus.
+        """
+        print("Running step 1...")
+
+        # perform skewness and variability tests
+        self.is_skewed = self.skewness_test_cytoplasm_texture()
+        self.is_variable = self.variability_test_formfactor()
+
+        # format results in a pretty table
+        print("Summary:\n")
+        print(
+            tabulate(
+                [
+                    ["Texture skewed?", bool_to_emoji(self.is_skewed)],
+                    ["Nucleus shape variable?", bool_to_emoji(self.is_variable)],
+                ],
+                headers=["Check", "Result"],
+                tablefmt="simple",
+            )
+        )
+
+        # set interpretation based on results
+        if self.is_skewed and self.is_variable:
+            interpretation = (
+                "Contamination detected! 🚨\n"
+                "Anomalous texture around nuclei detected and "
+                "nuclei segmentation is impacted.\n"
+                "Proceeding to step 2..."
+            )
+        elif self.is_skewed:
+            interpretation = (
+                "Contamination detected! 🚨\n"
+                "Anomalous texture around nuclei detected but "
+                "nuclei segmentation not clearly impacted.\n"
+                "Proceeding to step 2..."
+            )
+        elif self.is_variable:
+            interpretation = (
+                "Potential contamination detected! 🛑\n"
+                "This could indicate contamination, segmentation issues, "
+                "or an interesting phenotype.\n"
+                "Proceeding to step 2..."
+            )
+        else:
+            interpretation = "No indication of contamination. Plate appears clean 🫧!"
+
+        print("Interpretation:\n")
```

Comment:
Consider making this an f-string to avoid the extra print below.

Reply:
Thanks for catching! Updated!